modules/sdk-coin-stx/test/unit/stx.ts

```
@@ -341,4 +343,137 @@ describe('STX:', function () {
+    });
+
+    it('should build an unsigned transaction when public keys are passed', async function () {
+      const rootAddress = testData.COLD_WALLET_ROOT_ADDRESS;
+      nock(`https://api.testnet.hiro.so`)
+        .get(`/extended/v1/address/${rootAddress}/balances`)
+        .reply(200, testData.ACCOUNT_BALANCE_RESPONSE);
+      nock(`https://api.testnet.hiro.so`)
+        .get(`/extended/v1/address/${rootAddress}/nonces`)
+        .reply(200, testData.ACCOUNT_NONCE_RESPONSE);
+      nock(`https://api.testnet.hiro.so`, { allowUnmocked: true })
+        .post(`/v2/fees/transaction`, testData.FEE_ESTIMATION_REQUEST)
+        .reply(200, testData.FEE_ESTIMATION_RESPONSE);
+
+      const recoveryOptions: RecoveryOptions = {
+        backupKey: testData.COLD_WALLET_PUBLIC_KEY_INFO.BACKUP_KEY,
+        userKey: testData.COLD_WALLET_PUBLIC_KEY_INFO.USER_KEY,
+        rootAddress: rootAddress,
+        recoveryDestination: testData.DESTINATION_ADDRESS_WRW,
+        bitgoKey: testData.COLD_WALLET_PUBLIC_KEY_INFO.BITGO_PUB_KEY,
+      };
+      const response: RecoveryTransaction = await basecoin.recover(recoveryOptions);
+      response.should.have.property('txHex');
+      assert.deepEqual(response.txHex, testData.COLD_WALLET_UNSIGNED_SWEEP_TX_HEX, 'tx hex not matching!');
+    });
+
+    it('should throw invalid root address when root address is missing or invalid', async function () {
+      const recoveryOptions: RecoveryOptions = {
+        backupKey: testData.COLD_WALLET_PUBLIC_KEY_INFO.BACKUP_KEY,
+        userKey: testData.COLD_WALLET_PUBLIC_KEY_INFO.USER_KEY,
+        rootAddress: '',
+        recoveryDestination: testData.DESTINATION_ADDRESS_WRW,
+        bitgoKey: testData.COLD_WALLET_PUBLIC_KEY_INFO.BITGO_PUB_KEY,
+      };
+      await basecoin.recover(recoveryOptions).should.rejectedWith('invalid root address!');
+    });
+
+    it('should throw invalid root address when root address is missing or invalid', async function () {
```

should throw invalid destination address when destination address is missing or invalid

---

Updated